### PR TITLE
chore: Make Pi.Fintype computable

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---
I make a computable version of `List.rec` called `List.rec'` and this lets me make a bunch of things computable

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Multiset/Basic.lean [hunk 1]
```diff
@@ -177,22 +177,20 @@ variable {C : Multiset α → Sort _}
...
-      (fun hl _ ↦ by congr 1; exact Quot.sound hl)
+      (fun hl _ => by congr 1; exact Quotient.sound hl)
       (C_cons_heq _ _ ⟦_⟧ _)
 #align multiset.rec Multiset.rec
 
 /-- Companion to `Multiset.rec` with more convenient argument order. -/
 @[elab_as_elim]
-protected
-noncomputable -- Porting note: added
+protected -- Porting note: added
```
Comment (rwbarton) [inline Mathlib/Data/Multiset/Basic.lean:193]:
```suggestion
protected
```
& above

**Comment by eric-wieser** [comment]
How does this compare to #1720?

**Comment by ChrisHughes24** [comment]
I'll close it. I didn't realise there was already a PR making this change

**Comment by eric-wieser** [comment]
It was posted in the same thread you posted yours in :)

**Pull request closed without merging**